File: api_client.py
import requests

class APIClient:
    def __init__(self, base_url):
        self.base_url = base_url

    def login(self, email, password):
        url = f"{self.base_url}/auth/login"
        resp = requests.post(url, json={"email": email, "password": password})
        resp.raise_for_status()
        return resp.json()["token"]

    def get_profile(self, token):
        url = f"{self.base_url}/user/me"
        headers = {"Authorization": f"Bearer {token}"}
        resp = requests.get(url, headers=headers)
        resp.raise_for_status()
        return resp

    def update_name(self, token, new_name):
        url = f"{self.base_url}/user/name"
        headers = {"Authorization": f"Bearer {token}"}
        resp = requests.patch(url, json={"name": new_name}, headers=headers)
        resp.raise_for_status()
        return resp

    def check_user_exists(self, email):
        url = f"{self.base_url}/exist"
        resp = requests.post(url, json={"email": email})
        resp.raise_for_status()
        return resp

File: tests/src/test/java/APIClient.java
import io.restassured.response.Response;
import io.restassured.specification.RequestSpecification;

import static io.restassured.RestAssured.given;

public class APIClient {
    private final RequestSpecification spec;

    public APIClient(RequestSpecification spec) {
        this.spec = spec;
    }

    public Response register(User user) {
        Response response = given()
                .spec(spec)
                .body(user)
                .when()
                .post("/auth/register");
        user.setName(response.jsonPath().getString("user.name"));
        user.setToken(response.jsonPath().getString("token"));
        return response;
    }

    public Response exist(User user) {
        return given()
                .spec(spec)
                .body(user)
                .when()
                .post("/exist");
    }

    public Response login(User user) {
        return given()
                .spec(spec)
                .body(user)
                .when()
                .post("/auth/login");
    }

    public Response me(User user) {
        return given()
                .spec(spec)
                .header("Authorization", "Bearer " + user.getToken())
                .when()
                .get("user/me");
    }

    public Response changeName(User user, String newName) {
        String json = String.format("{\"name\": \"%s\"}", newName);
        Response response= given()
                .spec(spec)
                .header("Authorization", "Bearer " + user.getToken())
                .body(json)
                .when()
                .patch("user/name");
        user.setName(response.jsonPath().getString("user.name"));
        return response;
    }
}
